gpflow/likelihoods/likelihoods.py

```diff
@@ -69,92 +69,304 @@ def inv_probit(x):
+        """
+        expected_shape = tf.broadcast_dynamic_shape(tf.shape(F)[:-1], tf.shape(Y)[:-1])
+        tf.debugging.assert_equal(tf.shape(result), expected_shape)
 
-        and this object represents
+    def check_latent_dims(self, F):
+        """
+        Ensure that a tensor of latent functions F has latent_dim as right most dimension.
+        :param F: function evaluation Tensor, with shape [..., latent_dim]
+        """
+        tf.debugging.assert_shapes([(F, (..., self.latent_dim))])
 
-            p(y|f)
+    def check_data_dims(self, Y):
+        """
+        Ensure that a tensor of data Y has observation_dim as right most dimension.
+        :param Y: observation Tensor, with shape [..., observation_dim]
+        """
+        tf.debugging.assert_shapes([(Y, (..., self.observation_dim))])
 
-        then this method computes the predictive mean
+    def log_prob(self, F, Y):
+        """
+        The log probability density log p(Y|F)
+        :param F: function evaluation Tensor, with shape [..., latent_dim]
+        :param Y: observation Tensor, with shape [..., observation_dim]:
+        :return log pdf, with shape [...]
+        """
+        self.check_last_dims_valid(F, Y)
+        res = self._log_prob(F, Y)
+        self.check_return_shape(res, F, Y)
+        return res
 
-           \int\int y p(y|f)q(f) df dy
+    def _log_prob(self, F, Y):
+        raise NotImplementedError
 
-        and the predictive variance
+    def conditional_mean(self, F):
+        """
+        The conditional mean of Y|F
+        :param F: function evaluation Tensor, with shape [..., latent_dim]
+        :return mean [..., observation_dim]
+        """
+        self.check_latent_dims(F)
+        expected_Y = self._conditional_mean(F)
+        self.check_data_dims(expected_Y)
+        return expected_Y
 
-           \int\int y^2 p(y|f)q(f) df dy  - [ \int\int y^2 p(y|f)q(f) df dy ]^2
+    def _conditional_mean(self, F):
+        raise NotImplementedError
 
-        Here, we implement a default Gauss-Hermite quadrature routine, but some
-        likelihoods (e.g. Gaussian) will implement specific cases.
+    def conditional_variance(self, F):
         """
+        The conditional marginal variance of Y|F: [var(Y₁|F), ..., var(Yₖ|F)]
+        where K = observation_dim
+        :param F: function evaluation Tensor, with shape [..., latent_dim]
+        :return variance [..., observation_dim]
+        """
+        self.check_latent_dims(F)
+        var_Y = self._conditional_variance(F)
+        self.check_data_dims(var_Y)
+        return var_Y
 
-        def integrand(*X):
-            return self.conditional_variance(*X) + self.conditional_mean(*X) ** 2
+    def _conditional_variance(self, F):
+        """
+        The conditional marginal variance of Y|F: [var(Y₁|F), ..., var(Yₖ|F)]
+        :param F: function evaluation Tensor, with shape [..., latent_dim]
+        :return variance [..., observation_dim]
+        """
```

Comment:
```suggestion
```
the other implementation methods don't have a docstring either, and I think is clear from context - no need to duplicate